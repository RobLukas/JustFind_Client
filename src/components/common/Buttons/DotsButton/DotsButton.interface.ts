interface DotsButtonProps {
  handleClick(): void;
}

export default DotsButtonProps;
