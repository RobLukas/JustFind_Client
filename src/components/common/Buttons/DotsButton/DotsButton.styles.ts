import styled from 'styled-components';
import MoreHorizIcon from '@material-ui/icons/MoreHoriz';

const DotsButton = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-basis: 35px;
  height: 35px;
`;

const MoreHorizMuiIcon = styled(MoreHorizIcon)`
  display: block;
`;

export default {
  DotsButton,
  MoreHorizMuiIcon,
};
