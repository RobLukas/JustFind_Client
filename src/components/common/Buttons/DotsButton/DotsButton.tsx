import React, { FC } from 'react';
import IconButton from '@material-ui/core/IconButton';

import S from './DotsButton.styles';
import DotsButtonProps from './DotsButton.interface';

const DotsButton: FC<DotsButtonProps> = ({ handleClick }: DotsButtonProps) => {
  return (
    <S.DotsButton>
      <IconButton onClick={handleClick} size={'small'}>
        <S.MoreHorizMuiIcon />
      </IconButton>
    </S.DotsButton>
  );
};

export default DotsButton;
