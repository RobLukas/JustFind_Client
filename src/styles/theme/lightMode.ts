export default {
  name: 'Light Theme',
  styles: {
    primary: {
      background: '#f3f6f8',
      color: '#37474f',
    },
    secondary: {
      background: '#fff',
      color: '#99a1ab',
    },
    dropdown: {
      button: {
        hover: '#f3f6f8',
      },
    },
    scrollbar: {
      thumbBackground: '#c1c1c1',
      background: 'rgba(0,0,0,0.07)',
      hover: 'rgba(0,0,0,0.09)',
    },
    filter: {
      button: {
        color: '#777',
      },
      selected: {
        background: 'linear-gradient(to right, #ba68c8, #9c27b0)',
        color: '#fff',
      },
    },
    offer: {
      header: {
        title: '#fff',
        filter: {
          brightness: '1',
        },
        icons: {
          companyName: '#ff5252',
          companySize: '#fb8c00',
          companyEMPType: '#ab47bc',
          companySkill: '#66bb6a',
          new: '#448aff',
        },
      },
      desc: {
        seperateLine: {
          color: '#f3f6f8',
        },
        skillPoint: {
          selected: '#ff4081',
          unselected: '#393939',
        },
      },
    },
    offers: {
      newTitle: {
        background: '#d9ddfc',
        color: '#4256ef',
      },
      techTitle: {
        background: '#fff',
        border: '#e0e0e0',
      },
    },
  },
};
