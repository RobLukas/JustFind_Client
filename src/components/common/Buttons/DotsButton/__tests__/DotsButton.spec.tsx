import React from 'react';
import 'jest-styled-components';
import { cleanup, render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import lightMode from 'styles/theme/lightMode';
import DotsButton from '../DotsButton';

describe('<DotsButton />', () => {
  afterEach(cleanup);
  it('render correctly', () => {
    const handleClick = jest.fn();

    const { container } = render(
      <ThemeProvider theme={lightMode}>
        <DotsButton handleClick={handleClick} />
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
