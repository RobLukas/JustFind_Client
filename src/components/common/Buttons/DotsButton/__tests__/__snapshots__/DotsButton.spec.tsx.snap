// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DotsButton /> render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-basis: 35px;
  -ms-flex-preferred-size: 35px;
  flex-basis: 35px;
  height: 35px;
}

.c1 {
  display: block;
}

<div
  class="c0"
>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root c1"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
